docs/en/deploy/Running-Alluxio-On-a-Cluster.md

```diff
@@ -9,281 +9,350 @@ priority: 2
+
+```bash
+$ ./bin/alluxio-start.sh all SudoMount
+```
+
+This will start the master on the node you are running it on, and start all the workers on all the
+nodes specified in the `conf/workers` file.
+
+## Verify Alluxio Cluster
+
+To verify that Alluxio is running, visit `http://<alluxio_master_hostname>:19999` to see the status
+page of the Alluxio master.
 
-{% include Running-Alluxio-on-a-Cluster/run-tests.md %}
+Alluxio comes with a simple program that writes and reads sample files in Alluxio. Run the sample program with:
 
-**Note**: If you are using EC2, make sure the security group settings on the master node allows
- incoming connections on the alluxio web UI port.
+```bash
+$ ./bin/alluxio runTests
+```
 
-## Running Alluxio with High Availability
+# Alluxio HA Cluster
 
 High availability in Alluxio is based upon a multi-master approach, where multiple master processes
 are running in the system. One of these processes is elected the leader and is used by all workers
 and clients as the primary point of contact. The other masters act as standby, and use the shared
-journal to ensure that they maintain the same file system metadata as the leader. A standby master can
- rapidly take over in the event of the leader failing.
-
-If the leader fails, a new leader is automatically selected from the available standby masters and
-Alluxio proceeds as usual. Note that while the failover to a standby master happens clients may
-experience brief delays or transient errors.
+journal to ensure that they maintain the same file system metadata as the leader. The standby masters
+do not serve any client or worker requests.
 
-### Prerequisites
+If the leader master fails, a standby master will automatically chosen to take over and become the
+new leader. Once the new leader starts serving, Alluxio clients and workers proceed as usual. Note
+that during the failover to a standby master, clients may experience brief delays or transient errors.
 
-There are two prerequisites for setting up a Alluxio cluster with high availability:
-
-* [ZooKeeper](http://zookeeper.apache.org/)
-* A shared reliable under filesystem on which to place the journal
+## Prerequisites
 
-Alluxio requires ZooKeeper for leader selection. This ensures that there is at most one leader
-master for Alluxio at any given time.
+* Multiple master nodes, and 1 or more worker nodes
+* SSH login without password to all nodes. You can add a public ssh key for the host into
+`~/.ssh/authorized_keys`. See [this tutorial](http://www.linuxproblem.org/art_9.html) for more details.
+* A shared storage system to mount to Alluxio (accessible by all Alluxio nodes)
+* A [ZooKeeper](http://zookeeper.apache.org/) instance. Alluxio masters use ZooKeeper for leader election,
+and Alluxio clients and workers use ZooKeeper to inquire about the identity of the current leader master.
+* A shared reliable filesystem on which to place the journal. This shared filesystem must be
```

Comment:
the difference between "shared storage system" and "shared reliable filesystem" might be too subtle/confusing for users trying to figure out the pre-requisites. Can we be more specific about what qualifies?

Reply:
I added some recommendations.